### PR TITLE
Update Rollup Caps to allow unknown fields

### PR DESCRIPTION
**Pull request by hub-cap**
This commit ensures that the parts of rollup caps that can allow unknown
fields will allow them. It also modifies the test such that we can use
the features we need for disallowing fields in spots where they would
not be allowed.

Relates #36938

### CODE REVIEW

File: client/rest-high-level/src/main/java/org/elasticsearch/client/rollup/GetRollupCapsResponse.java [hunk 1]
```diff
@@ -39,7 +39,7 @@ public GetRollupCapsResponse(final Map<String, RollableIndexCaps> jobs) {
     }
 
     public static GetRollupCapsResponse fromXContent(final XContentParser parser) throws IOException {
-        Map<String, RollableIndexCaps> jobs = new HashMap<>();
+        Map<String, RollableIndexCaps> jobs = new TreeMap<>();
```
Comment (dakrone) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/rollup/GetRollupCapsResponse.java]: Why does this now care about insert order?

File: client/rest-high-level/src/main/java/org/elasticsearch/client/rollup/RollupJobCaps.java [hunk 2]
```diff
@@ -62,7 +59,7 @@
         PARSER.declareString(ConstructingObjectParser.constructorArg(), ROLLUP_INDEX);
         PARSER.declareString(ConstructingObjectParser.constructorArg(), INDEX_PATTERN);
         PARSER.declareNamedObjects(ConstructingObjectParser.constructorArg(),
-            (p, c, name) -> new Tuple<>(name, RollupFieldCaps.fromXContent(p)), FIELDS);
+            (p, c, name) -> new AbstractMap.SimpleEntry<>(name, RollupFieldCaps.fromXContent(p)), FIELDS);
```
Comment (dakrone) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/rollup/RollupJobCaps.java]: Why not use `Tuple` here and then use `v1/v2` in the `Collectors.toMap` here? I'm just curious about why the switch

**Review by dakrone — COMMENTED** [review]
I left two questions for this one

File: client/rest-high-level/src/main/java/org/elasticsearch/client/rollup/GetRollupCapsResponse.java (hunk 1 above)
Comment (hub-cap) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/rollup/GetRollupCapsResponse.java]: i did this on accident, good catch. it is a test sorting order issue iirc that i need to fix.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/rollup/RollupJobCaps.java (hunk 2 above)
Comment (hub-cap) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/rollup/RollupJobCaps.java]: ok this has been addressed in my new commit <3

**Review by dakrone — APPROVED** [review]
LGTM, thanks Baz

**Review by polyfractal — APPROVED** [review]
TreeMap was the only thing I noticed, so LGTM2 :)

Thanks @hub-cap!

**Pull request merged**